src/backend/PortabilityService.ConfigurationSource/ConfigurationSource.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Extensions.Configuration.Memory;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace PortabilityService.ConfigurationSource
+{
+    public class PortabilityServiceConfigurationSource : MemoryConfigurationSource
+    {
+        private const string DefaultConfigurationServiceUrl = "http://localhost:1443";
```

Comment:
Moved this to constants.cs file. I like having the hard-coded URL because this is what the configurationservice is configured for when doing local development. This should come from the environment outside of local development.

Reply:
When we startup the configuration service, it should be setting the environment variable, `PortabilityConfigurationServiceUrl`.

If we want to ensure that all our consuming projects have the environment variable defined, we can move that logic into our `Directory.Build.Props` or `Directory.Build.targets` for development.

The downside to having this logic compiled into a library, is that if we decide to change the localhost:1443 to like localhost:1000 in the service, we will be forced to recompile that library, find the string in another location and replace that.

Reply:
I was thinking more about this, and msbuild doesn't allow by default to set Environment Variables from your msbuild script. I think it makes sense because you wouldn't want to run MSBuild and have it modify your user environment.

I was wondering if it is better to have something like [PortabilityServiceConfigurationSettings.Development.json](https://github.com/Microsoft/dotnet-apiport/blob/service/src/backend/PortabilityService.ConfigurationService/PortabilityServiceConfigurationSettings.Development.json) in the root of our repository and import them like we do in [Directory.Build.targets](https://github.com/Microsoft/dotnet-apiport/blob/service/Directory.Build.targets#L37-L39) where we include it in an itemgroup and use it only for development.

This would remove the entire use of a environment variable.

What do you think @twsouthwick?

Reply:
This is tricky as it's a bootstrapping problem for configuration... Which aspects here are potentially changeable? The host (`localhost`) and/or port (`1443`)? What about conditioning it off of `IHostingEnvironment.IsDevelopment()`?

Reply:
Updated to remove the constant and throw if environment can't be set as discussed off line